### PR TITLE
HappiTails Project

### PR DESCRIPTION
**Pull request by annehomann**
Stuck on the last task of Phase 4 :/

### CODE REVIEW

**Comment by annehomann** [comment]
Completed project with much help from Joel

File: students/anne_homann/happitails_app/client.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
+class Client
+  attr_accessor :name, :children, :age, :pets
+
+  def initialize(name, children, age, pets)
+    @name = name
+    @children = children
+    @age = age
+    @pets = pets
+  end
+
```
Comment (mathildathompson) [inline students/anne_homann/happitails_app/client.rb:10]: @pets need to be an empty hash to store key value pairs in. 

You need to chance the initialize method to this: 

```
def initialize(name, children, age)
  @name = name
  @children = children
  @age = age
  @pets = {}
end
```

Let me know if this does not make sense!

File: students/anne_homann/happitails_app/main.rb [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+
+
+        animal = shelter.animals[adoptionAnimal]
+        shelter.animals.delete(adoptionAnimal)
+        puts "#{adopting_client} has adopted #{adoptionAnimal}! Thanks #{adopting_client}!"
+        puts "___________________________________________________"
+        # adoptionNew = shelter.animals.delete(adoptionAnimal)
+        
+        shelter.clients[adopting_client].pets[adoptionAnimal] = animal
+
```
Comment (mathildathompson) [inline students/anne_homann/happitails_app/main.rb:121]: Lines 89 -120 would look nicer like this: 

```
    puts "Choose a client #{shelter.clients.keys.join(',')} "
    # puts "Which client wants to adopt? "
    adopting_client = gets.chomp

    puts "___________________________________________________"
    puts "Which animal do you want to adopt #{shelter.animals.keys.join(',')}"
    adoptionAnimal = gets.chomp

    animal = shelter.animals[adoptionAnimal]
    shelter.animals.delete(adoptionAnimal)
    puts "#{adopting_client} has adopted #{adoptionAnimal}! Thanks #{adopting_client}!"
    puts "___________________________________________________"

    shelter.clients[adopting_client].pets[adoptionAnimal] = animal
```

Try and implement the same refactoring for putting an animal up for adoption below.

File: students/anne_homann/happitails_app/main.rb [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    
+end
+            
+
+
+
+
+
+#binding.pry
```
Comment (mathildathompson) [inline students/anne_homann/happitails_app/main.rb:176]: Overall an excellent attempt at this exercise I am very impressed - let me know if you have any questions!

**Pull request merged**